src/test/common/test_blkdev.cc

```diff
@@ -91,3 +91,19 @@ TEST_F(BlockDevTest, is_rotational)
   EXPECT_FALSE(sda.is_rotational());
   EXPECT_TRUE(sdb.is_rotational());
 }
+
+TEST(blkdev, _decode_model_enc)
+{
+
+  const char *foo[][2] = {
```

Comment:
could use an array of pair to utilize the structured binding introduced by C++17, like
```
  const pair<string, string> foo[2] = {
    { "WDC\\x20WDS200T2B0A-00SM50\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20\\x20",
      "WDC_WDS200T2B0A-00SM50" },
  };

  for (auto& [model,decoded] : foo) {
    ASSERT_EQ(decoded, _decode_model_enc(model));
  }
```